### PR TITLE
Update branching.md

### PR DESCRIPTION
**Pull request by danieltprice**
Draft content for branching.

https://websitemain62807-dpricebranchingintrocontent.gtsb.io/docs/conceptual-guides/branching/

### CODE REVIEW

File: content/docs/conceptual-guides/branching.md [hunk 1]
```diff
@@ -1,21 +1,115 @@
...
+2. Click **New Branch**.
+
+3. Enter a name for the branch.
+
+3. Select the Neon project or brach that you want to branch from.
+
+4. Select the type of branch you want to create. 
+    - **Head**: Branch the current state of the parent project. The branch is created with all of the parent project data.
+    - **Time**: Branch a specific point in time. The branch is created with the project data as it existed at the specified date and time.
+    - **LSN**: Branch from a specified Log Sequence NUmber (LSN). The branch is created with the project data as it existed at the specified LSN.
```
Comment (raoufchebri) [inline content/docs/conceptual-guides/branching.md:86]: Should the`u` in `NUmber` be lowercase?

**Review by raoufchebri — COMMENTED** [review]
The content is great. I would change the structure though. It seems that the user would scroll down a lot to get to how to create a branch. We can move the "Create a branch" part up. wdyt?

**Comment by stepashka** [comment]
I'll review too

File: content/docs/conceptual-guides/branching.md (hunk 1 above)
Comment (stepashka) [inline content/docs/conceptual-guides/branching.md:86]:
```suggestion
    - **LSN**: Branch from a specified Log Sequence Number (LSN). The branch is created with the project data as it existed at the specified LSN.
```

Comment (danieltprice) [reply content/docs/conceptual-guides/branching.md]: Fixed!

**Review by raoufchebri — DISMISSED** [review]

File: content/docs/conceptual-guides/branching.md [hunk 2]
```diff
@@ -1,21 +1,188 @@
...
+- **Staging**
+    - Create a staging database by branching your production data
+    - Create a branch for every pull request in your CI/CD pipeline
+			
+- **Backup**
+    -	Quickly and easily create backup branches
+    -	Instantly restore a previous state by branching from a previously created backup branch
+    -	Name backup branches according to the time they were created for convenient point-in-time restore (PITR)
+	
+- **Replication**
```
Comment (stepashka) [inline content/docs/conceptual-guides/branching.md:44]: drop?

Comment (danieltprice) [reply content/docs/conceptual-guides/branching.md]: dropped.

**Comment by danieltprice** [comment]
> The content is great. I would change the structure though. It seems that the user would scroll down a lot to get to how to create a branch. We can move the "Create a branch" part up. wdyt?

I moved the how to content to the getting started topic. 
https://websitemain62807-dpricebranchinggetstarted.gtsb.io/docs/get-started-with-neon/get-started-branching/

File: content/docs/conceptual-guides/branching.md [hunk 3]
```diff
@@ -1,21 +1,56 @@
...
 
-You can use a branch to:
+## What is a branch?
 
-- Run potentially destructive queries without impacting your main branch
-- Run time travel queries against historical state
-- Run a set of queries with separate resources to avoid impacting your application
-- Tag and name the current moment for PITR convenience or ML model training repeatability
-- Run tests against a branch from production data
+A branch is a copy-on-write clone of your data. You can create a branch from a current or past state. For example, you can create a branch that includes all data up to the current point in time or up to a past point in time or [Log Sequence Number (LSN)](../../reference/glossary#lsn).
```
Comment (stepashka) [inline content/docs/conceptual-guides/branching.md:16]: not sure the link `../../reference/glossary#lsn` resolves for me!
do we have that header in glossary?

File: content/docs/conceptual-guides/branching.md [hunk 4]
```diff
@@ -1,21 +1,56 @@
...
+
+### Testing
+
+Branching enables testers to use the most recent production data. Testers can create branches for testing schema changes, validating new queries, or testing potentially destructive queries before deploying them into production. A branch is isolated from its parent branch but has all of the parent branch's data up to the point of branch creation, which eliminates the effort involved in hydrating a database. Tests can also run on separate branches in parallel, with each branch having dedicated compute resources.
+
+Another testing scenario enabled by branching is tracking down corruption or data quality issue. For example, you can create and dispose of as many point-in-time database branches as necessary to determine when the corruption or data quality issue first appeared.
```
Comment (stepashka) [inline content/docs/conceptual-guides/branching.md:36]:
```suggestion
Another testing scenario enabled by branching is tracking down corruption or data quality issue. For example, you can create and dispose of multiple point-in-time database branches to determine when the corruption or data quality issue first appeared.
```

the amount of endpoints one can have simultaneously will be limited.
there is also a limit on the amount of branches you can have.

File: content/docs/conceptual-guides/branching.md [hunk 5]
```diff
@@ -1,21 +1,56 @@
...
+
+### Staging
+
+With Neon's branching capabilities, you can create a staging database by branching your production database. Using the Neon API, you can automate branch creation for every pull request in your CI/CD pipeline.
```
Comment (stepashka) [inline content/docs/conceptual-guides/branching.md:40]:
```suggestion
With Neon's branching capabilities, you can create a staging database by branching your production database. Using the Neon API, you can automate branch creation creating a database branch based on the staging branch for every pull request in your CI/CD pipeline.
```

**Review by stepashka — COMMENTED** [review]
this PR adds the images that are not used in this article anymore. consider disposing of them.

**Review by stepashka — APPROVED** [review]
let's ship it and then we can iterate further

**Pull request merged**